pathway-analysis/qusage_meta_analysis.Rmd

```diff
@@ -0,0 +1,429 @@
+We 'functionalize' it to keep from repeating ourselves. 
+For an example where we do the conversion without using a custom function, see
+[`qusage_single_dataset`](./qusage_single_dataset.nb.html) or 
+[`qusage_replicate_vignette`](./qusage_replicate_vignette.nb.html).
+
+```{r}
+convert_ensembl_to_entrez_mat <- function(exprs_df) {
+  # Given a data.frame that contains human gene expression values and a column 
+  # (named ENSEMBL) that contains Ensembl gene IDs, return an expression matrix
+  # where the rownames are Entrez IDs. In the case of duplicate Entrez 
+  # identifiers, we summarize to the mean value for an Entrez ID. Mapping
+  # is performed with AnnotationDbi::mapIDs.
+  # 
+  # Args:
+  #   exprs_df: A data.frame of gene expression data. The first column should be
+  #             named ENSEMBL and contain Ensembl gene IDs. Rows are genes; 
+  #             columns are samples.
+  # Returns:
+  #   An expression matrix where the rownames are Entrez gene IDs
+  
+  `%>%` <- dplyr::`%>%`
+  require(org.Hs.eg.db)
+  
+  # error-handling: we need an ENSEMBL column
+  if (!("ENSEMBL" %in% colnames(exprs_df))) {
+    stop("'ENSEMBL' column expected in exprs_df")
+  }
+  
+  # using the default behavior for 1:many mappings, where only the first one is 
+  # selected
+  entrez_mappings <- mapIds(org.Hs.eg.db, keys = exprs_df$ENSEMBL, 
+                            column = "ENTREZID", keytype = "ENSEMBL")
+  
+  # if this is not returned in the same order as the keys for some reason, stop
+  if (!all.equal(names(entrez_mappings), exprs_df$ENSEMBL)) {
+    stop("Something happened to the gene order!")
+  }
+  
+  # annotation with Entrez IDs
+  entrez_exprs_df <- exprs_df %>%
+    # add a new column that contains the Entrez IDs
+    # this gets added as the last column
+    dplyr::mutate(ENTREZID = entrez_mappings) %>%
+    # drop the Ensembl gene IDs
+    dplyr::select(-ENSEMBL) %>%
+    # reorder such that the Entrez IDs are in the first column
+    dplyr::select(ENTREZID, dplyr::everything()) %>%
+    # drop any genes without an Entrez ID
+    dplyr::filter(!is.na(ENTREZID))
+  
+  # if there are any duplicate Entrez gene IDs, collapse to the mean value
+  if (any(duplicated(entrez_exprs_df$ENTREZID))) {
+    message("Collapsing to mean value...")
+    entrez_exprs_df <- entrez_exprs_df %>%
+      dplyr::group_by(ENTREZID) %>%
+      dplyr::summarise_all(mean)
+  }
+  
+  # expression matrix where the rownames are the gene identifiers
+  exprs_mat <- entrez_exprs_df %>%
+    tibble::column_to_rownames("ENTREZID") %>%
+    as.matrix()
+
+}
+```
+
+### Gene sets
+
+`qusage` allows you to read in gene sets that are in the [GMT format](http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29).
+[MSigDB](http://software.broadinstitute.org/gsea/msigdb) offers genesets in this format.
+
+For more information or recommendations about gene sets, [see the Gene Sets
```

Comment:
```suggestion
For more information or recommendations about gene sets, [see the 'Gene Sets'
```
Maybe something to make more clear it's a headline and that's why you are capitalizing it? IDK

Reply:
Made it bold